File: app/src/models/Registration.ts
import type { DbModel, User } from './index'
import type { RegistrationId, SeasonId, UserId } from './ids'

export interface Registration extends DbModel {
  user_id: UserId
  season_id: SeasonId
  user: User
  questions: Array<Question>
}

export interface Question extends DbModel {
  registration_id: RegistrationId
  text: string
  answer: string
  render: string
}

export type { RegistrationId } from './ids'

File: internal/models/registration.go
package models

type Registration struct {
	DbModel
	SeasonID  uint `json:"season_id"`
	UserID    uint
	User      User       `json:"user"`
	Questions []Question `type:"questions"`
}

type Question struct {
	DbModel
	RegistrationID uint
	Text           string `json:"text"`
	Answer         string `json:"answer"`
	Render         string `json:"render"`
}
